[mono] Linux arm64 AOT build failing on perf jobs

Since 20th of December (https://dev.azure.com/dnceng/internal/_build/results?buildId=2605880&view=results), linux arm64 AOT builds on perf runs are failing with 
```
2024-12-27T08:08:32.3117042Z   Running 'PYTHONPATH=/__w/1/s/.packages/runtime.linux-x64.microsoft.netcore.runtime.mono.llvm.libclang/19.1.0-alpha.1.24569.1/tools/bindings/python python3 /__w/1/s/src/mono/mono/offsets/offsets-tool.py --abi=aarch64-linux-gnu --targetdir="/__w/1/s/artifacts/obj/mono/linux.arm64.Release" --monodir="/__w/1/s/src/mono" --nativedir="/__w/1/s/src/native" --outfile="/__w/1/s/artifacts/obj/mono/linux.arm64.Release/cross/offsets-aarch-linux-gnu.h" --libclang="/__w/1/s/.packages/runtime.linux-x64.microsoft.netcore.runtime.mono.llvm.libclang/19.1.0-alpha.1.24569.1/tools/libclang.so" --libclang-headers="/__w/1/s/.packages/runtime.linux-x64.microsoft.netcore.runtime.mono.llvm.libclang/19.1.0-alpha.1.24569.1/tools/include" --prefix="/crossrootfs/arm64/usr/lib/gcc/aarch64-linux-gnu/5" --sysroot="/crossrootfs/arm64"'
2024-12-27T08:08:33.1176807Z   provided path via --prefix ("/crossrootfs/arm64/usr/lib/gcc/aarch64-linux-gnu/5") doesn't exist.
2024-12-27T08:08:33.1252035Z /__w/1/s/src/mono/mono.proj(1000,5): error MSB3073: The command "python3 /__w/1/s/src/mono/mono/offsets/offsets-tool.py --abi=aarch64-linux-gnu --targetdir="/__w/1/s/artifacts/obj/mono/linux.arm64.Release" --monodir="/__w/1/s/src/mono" --nativedir="/__w/1/s/src/native" --outfile="/__w/1/s/artifacts/obj/mono/linux.arm64.Release/cross/offsets-aarch-linux-gnu.h" --libclang="/__w/1/s/.packages/runtime.linux-x64.microsoft.netcore.runtime.mono.llvm.libclang/19.1.0-alpha.1.24569.1/tools/libclang.so" --libclang-headers="/__w/1/s/.packages/runtime.linux-x64.microsoft.netcore.runtime.mono.llvm.libclang/19.1.0-alpha.1.24569.1/tools/include" --prefix="/crossrootfs/arm64/usr/lib/gcc/aarch64-linux-gnu/5" --sysroot="/crossrootfs/arm64"" exited with code 1.
2024-12-27T08:08:33.1302978Z ##[error]src/mono/mono.proj(1000,5): error MSB3073: (NETCORE_ENGINEERING_TELEMETRY=Build) The command "python3 /__w/1/s/src/mono/mono/offsets/offsets-tool.py --abi=aarch64-linux-gnu --targetdir="/__w/1/s/artifacts/obj/mono/linux.arm64.Release" --monodir="/__w/1/s/src/mono" --nativedir="/__w/1/s/src/native" --outfile="/__w/1/s/artifacts/obj/mono/linux.arm64.Release/cross/offsets-aarch-linux-gnu.h" --libclang="/__w/1/s/.packages/runtime.linux-x64.microsoft.netcore.runtime.mono.llvm.libclang/19.1.0-alpha.1.24569.1/tools/libclang.so" --libclang-headers="/__w/1/s/.packages/runtime.linux-x64.microsoft.netcore.runtime.mono.llvm.libclang/19.1.0-alpha.1.24569.1/tools/include" --prefix="/crossrootfs/arm64/usr/lib/gcc/aarch64-linux-gnu/5" --sysroot="/crossrootfs/arm64"" exited with code 1.
```

The range of dotnet/runtime commits when the regression was introduced https://github.com/dotnet/runtime/compare/cf4d2b0594a7af4a5ff39ffd4fe5cfb0c5577a3d...57ab984bd0dbbacd02315d41e09144d2823e9475

I wonder if the update to newer build images from https://github.com/dotnet/runtime/pull/110198 is triggering:

`2024-12-27T08:08:33.1176807Z   provided path via --prefix ("/crossrootfs/arm64/usr/lib/gcc/aarch64-linux-gnu/5") doesn't exist.`

and we might need to change https://github.com/dotnet/runtime/blob/c969265440b74e95429fb513e6d940d6c9bc2fd3/src/mono/mono.proj#L817.

Last non-failing build: https://dev.azure.com/dnceng/internal/_build/results?buildId=2605380

--- a/src/mono/mono.proj
+++ b/src/mono/mono.proj
@@ -814,7 +814,6 @@ JS_ENGINES = [NODE_JS]
       <MonoAotAbi>aarch64-linux-gnu</MonoAotAbi>
       <MonoAotOffsetsFile>$(MonoObjCrossDir)offsets-aarch-linux-gnu.h</MonoAotOffsetsFile>
       <MonoAotOffsetsPrefix>$(MonoCrossDir)/usr/lib/gcc/aarch64-linux-gnu/7</MonoAotOffsetsPrefix>
-      <MonoAotOffsetsPrefix Condition="'$(Platform)' == 'arm64'">$(MonoCrossDir)/usr/lib/gcc/aarch64-linux-gnu/5</MonoAotOffsetsPrefix>
     </PropertyGroup>
 
     <PropertyGroup Condition="'$(MonoUseLLVMPackage)' == 'true'">

